### DISCUSSION TITLE
About Widget Life Cycle (mount/unmount)

### DISCUSSION THREAD

**Discussion by iacore** [fyne-io/fyne#3376] [Q&A] [answered]
Is there a way to run some function when a widget becomes visible? This is called "component life cycle" in Web UI's term.

**Comment by Jacalz** [comment, marked as answer]
There is not really an API for that at the moment. A hack (not technically meant to be used in that way and not guaranteed to happen exactly at that time) the CreateRenderer method of the widget will be called more or less when it gets visible.

You can extend the widget and replace the CreateRenderer() method with your own (similar to  https://developer.fyne.io/extend/extending-widgets). Just make sure that you return the result of the parent's CreateRenderer() method, otherwise it will not behave as expected (or not work at all).

**Reply by iacore** [reply]
Thanks! What about the opposite of `onMount`, something like `onDestroy`?

**Reply by Jacalz** [reply]
I don't know what you mean with `onMount` but I assume it is what you were referring to in the question. There is not quite a good way to hack together some sort of opposite to it.

**Reply by PaulWaldo** [reply]
I think the `Destroy` method on your extended Renderer might be what you are looking for @locriacyber 

```
type WidgetRenderer interface {
	// Destroy is for internal use.
	Destroy()
	// Layout is a hook that is called if the widget needs to be laid out.
	// This should never call Refresh.
	Layout([Size](https://pkg.go.dev/fyne.io/fyne/v2@v2.2.3#Size))
	// MinSize returns the minimum size of the widget that is rendered by this renderer.
	Min[Size](https://pkg.go.dev/fyne.io/fyne/v2@v2.2.3#Size)() Size
	// Objects returns all objects that should be drawn.
	Objects() [][CanvasObject](https://pkg.go.dev/fyne.io/fyne/v2@v2.2.3#CanvasObject)
	// Refresh is a hook that is called if the widget has updated and needs to be redrawn.
	// This might trigger a Layout.
	Refresh()
}
```
